src/sql-parser/src/parser.rs

```diff
@@ -3935,12 +3948,26 @@ impl<'a> Parser<'a> {
         // parse optional column list (schema)
         let (columns, constraints) = self.parse_columns(Mandatory)?;
 
+        let with_options = if self.parse_keyword(WITH) {
+            self.expect_token(&Token::LParen)?;
+            let o = if matches!(self.peek_token(), Some(Token::RParen)) {
+                vec![]
+            } else {
+                self.parse_comma_separated(Parser::parse_table_option)?
+            };
+            self.expect_token(&Token::RParen)?;
+            o
+        } else {
+            vec![]
+        };
```

Comment:
This appears to be where PostgreSQL puts the `WITH` clause as well for `CREATE TABLE`, so LGTM from a SQL council perspective.